pylint/checkers/imports.py

```diff
@@ -969,6 +976,14 @@ def _wildcard_import_is_allowed(self, imported_module):
             and "__all__" in imported_module.locals
         )
 
+    def _check_toplevel(self, node):
+        if isinstance(node.parent, (astroid.FunctionDef, astroid.ClassDef)):
```

Comment:
If you use `node.scope()` here, it will traverse up the syntax tree until it finds the node that defines the scope (https://astroid.readthedocs.io/en/latest/api/base_nodes.html#astroid.node_classes.NodeNG.scope). Anything other than a `Module` would mean that the message should be raised. This should even account for cases like the following where we wouldn't want to raise a message:

```python
if some_condition:
    import this
else:
    import that
```

and cases like the following where we would want to raise the message:

```python
def func():
    if some_condition:
        import this
    ...
```

Reply:
I knew there was something like that, but I couldn't remember the name. This should be fixed now.